tutorials/09-set-up-physics-engine-tpe.md

```diff
@@ -0,0 +1,246 @@
+in `EntityManagementFeatures` for TPE.
+
+Basically, \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+has a subclass \ref ignition::physics::Feature::Engine "Engine" defining
+`ConstructEmptyWorld` member function. This feature will construct an empty
+simulation world from using the physics engine API. We will display the feature
+implementation here for convenience:
+- [EntityManagementFeatures.hh](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/EntityManagementFeatures.hh):
+
+```cpp
+#include <string>
+
+#include <ignition/physics/ConstructEmpty.hh>
+#include <ignition/physics/Shape.hh>
+#include <ignition/physics/GetEntities.hh>
+#include <ignition/physics/RemoveEntities.hh>
+#include <ignition/physics/Implements.hh>
+
+#include "Base.hh"
+
+namespace ignition {
+namespace physics {
+namespace tpeplugin {
+
+struct EntityManagementFeatureList : FeatureList<
+  GetEngineInfo,
+  GetWorldFromEngine,
+  GetModelFromWorld,
+  GetLinkFromModel,
+  GetShapeFromLink,
+  RemoveEntities,
+  ConstructEmptyWorldFeature,
+  ConstructEmptyModelFeature,
+  ConstructEmptyLinkFeature,
+  CollisionFilterMaskFeature
+> { };
+
+class EntityManagementFeatures :
+  public virtual Base,
+  public virtual Implements3d<EntityManagementFeatureList>
+{
+  ... // other features' member functions
+
+  // ----- Construct empty entities -----
+  public: Identity ConstructEmptyWorld(
+    const Identity &_engineID, const std::string &_name) override;
+
+  ... // other features' member functions
+};
+
+}
+}
+}
+```
+
+Together with other \ref ignition::physics::Feature "Features", the \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+is included in `EntityManagementFeatureList` "FeatureList" to declare the related
+features for entity management purpose.
+
+The `EntityManagementFeatures` "FeatureList" here inherits from:
+- [Base](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/Base.hh)
+class for foundation metadata definitions of Models, Joints, Links, and Shapes objects
+of TPE to provide easy access to [tpelib](https://github.com/ignitionrobotics/ign-physics/tree/main/tpe/lib)
+structures in the TPE library.
+- \ref ignition::physics::Implements3d "Implements3d" for implementing the
+custom feature with \ref ignition::physics::FeaturePolicy3d "FeaturePolicy3d"
+("FeaturePolicy" of 3 dimensions and scalar type `double`).
+
+Based on each \ref ignition::physics::Feature "Features" class definition, we
+will override the corresponding functions of all the class defined in
+`EntityManagementFeatureList` (the feature definitions can be found [here](https://ignitionrobotics.org/api/physics/3.0/classes.html)).
+Here we show the overriding of `ConstructEmptyWorld` member function of \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature".
```

Comment:
this is where they use the physics engine implementation to implement this member function